x-pack/plugins/security_solution/server/lib/detection_engine/rules/update_prepacked_rules.ts

```diff
@@ -6,123 +6,177 @@
+ * if we don't limit its number of connections as we increase the number of rules that can be
+ * installed at a time.
+ *
+ * Lastly, we saw weird issues where Chrome on upstream 408 timeouts will re-call the REST route
+ * which in turn could create additional connections we want to avoid.
+ *
+ * See file import_rules_route.ts for another area where 50 was chosen, therefore I chose
+ * 50 here to mimic it as well. If you see this re-opened or what similar to it, consider
+ * reducing the 50 above to a lower number.
+ *
+ * See the original ticket here:
+ * https://github.com/elastic/kibana/issues/94418
+ */
+export const UPDATE_CHUNK_SIZE = 50;
+
+/**
+ * Updates the prepackaged rules given a set of rules and output index.
+ * This implements a chunked approach to not saturate network connections and
+ * avoid being a "noisy neighbor".
+ * @param alertsClient Alerting client
+ * @param savedObjectsClient Saved object client
+ * @param rules The rules to apply the update for
+ * @param outputIndex The output index to apply the update to.
+ */
 export const updatePrepackagedRules = async (
   alertsClient: AlertsClient,
   savedObjectsClient: SavedObjectsClientContract,
   rules: AddPrepackagedRulesSchemaDecoded[],
   outputIndex: string
 ): Promise<void> => {
-  await Promise.all(
-    rules.map(async (rule) => {
-      const {
-        author,
-        building_block_type: buildingBlockType,
-        description,
-        event_category_override: eventCategoryOverride,
-        false_positives: falsePositives,
-        from,
-        query,
-        language,
-        license,
-        saved_id: savedId,
-        meta,
-        filters: filtersObject,
-        rule_id: ruleId,
-        index,
-        interval,
-        max_signals: maxSignals,
-        risk_score: riskScore,
-        risk_score_mapping: riskScoreMapping,
-        rule_name_override: ruleNameOverride,
-        name,
-        severity,
-        severity_mapping: severityMapping,
-        tags,
-        to,
-        type,
-        threat,
-        threshold,
-        threat_filters: threatFilters,
-        threat_index: threatIndex,
-        threat_query: threatQuery,
-        threat_mapping: threatMapping,
-        threat_language: threatLanguage,
-        concurrent_searches: concurrentSearches,
-        items_per_search: itemsPerSearch,
-        timestamp_override: timestampOverride,
-        references,
-        version,
-        note,
-        anomaly_threshold: anomalyThreshold,
-        timeline_id: timelineId,
-        timeline_title: timelineTitle,
-        machine_learning_job_id: machineLearningJobId,
-        exceptions_list: exceptionsList,
-      } = rule;
+  const ruleChunks = chunk(UPDATE_CHUNK_SIZE, rules);
+  for (const ruleChunk of ruleChunks) {
+    const rulePromises = createPromises(alertsClient, savedObjectsClient, ruleChunk, outputIndex);
+    await Promise.all(rulePromises);
+  }
```

Comment:
It's time to push bulk CRUD APIs again it seems :) https://github.com/elastic/kibana/issues/53144